packages/sitecore-jss-dev-tools/src/templating/react/generate-component-builder.ts

```diff
@@ -0,0 +1,63 @@
+import path from 'path';
+import fs from 'fs';
+import { getComponentBuilderTemplate } from './templates/component-builder';
+import { PackageDefinition, getComponentList } from '../components';
+import { watchItems } from '../utils';
+
+// Default destination path for component builder
+const componentBuilderOutputPath = 'src/temp/componentBuilder.js';
+
+/**
+ * Generate component builder based on provided settings
+ * @param {Object} settings settings for component builder generation
+ * @param {string} settings.componentRootPath path to components root
+ * @param {PackageDefinition[]} settings.packages list of packages to include in component builder
+ * @param {boolean} settings.watch whether to watch for changes to component builder sources
+ */
+export function generateComponentBuilder({
+  componentRootPath = 'src/components',
+  packages,
+  watch,
+}: {
+  componentRootPath?: string;
+  packages?: PackageDefinition[];
+  watch?: boolean;
+}) {
+  if (watch) {
+    watchComponentBuilder(componentRootPath, packages);
+  } else {
+    writeComponentBuilder(componentRootPath, packages);
+  }
+}
+
+/**
+ * Watch for changes to component builder sources
+ * @param {string} componentRootPath root path to components
+ * @param {PackageDefinition[]} [packages] packages to include in component builder
+ */
+export function watchComponentBuilder(componentRootPath: string, packages?: PackageDefinition[]) {
+  console.log(`Watching for changes to component builder sources in ${componentRootPath}...`);
+
+  watchItems([componentRootPath], writeComponentBuilder.bind(null, componentRootPath, packages));
```

Comment:
Should component builder be written once, initially, before we launch watch on the component folder? With this current way it will only be written on changes to root path, if it wasn't present.

Reply:
No, it shouldn't be written during "watch" stage, because, for instance, we run (start-connected)[https://github.com/Sitecore/jss/blob/feature/584599-dev/packages/create-sitecore-jss/src/templates/react/package.json#L97]), it will run "bootstrap" and "watch-components", so during the bootstrap stage we need to write builder, during "watch" - only watch